editoast/src/core/simulation.rs

```diff
@@ -306,14 +306,15 @@ pub struct ReportTrain {
 pub struct CompleteReportTrain {
     #[serde(flatten)]
     pub report_train: ReportTrain,
-    pub signal_sightings: Vec<SignalSighting>,
+    pub signal_critical_positions: Vec<SignalCriticalPosition>,
     pub zone_updates: Vec<ZoneUpdate>,
     pub spacing_requirements: Vec<SpacingRequirement>,
     pub routing_requirements: Vec<RoutingRequirement>,
 }
 
 #[derive(Debug, Clone, PartialEq, Hash, Serialize, Deserialize, ToSchema)]
-pub struct SignalSighting {
+/// First position (space and time) along the path where given signal must be free
```

Comment:
@woshilapin (reply to https://github.com/OpenRailAssociation/osrd/pull/9661#pullrequestreview-2457946812): critical position is exposed in openapi (even though this part is unread by front, only exchanged between core and editoast).
And it's indeed better than signal-sighting IMO because it's not always signal-sighting.

Would it be better with this change?
```suggestion
/// First position (space and time) along the path where given signal must be free (sighting time or closed-signal stop ending)

Note that it is already documented a bit [here](https://github.com/OpenRailAssociation/osrd/pull/9661/files#diff-042f91ba14bffb49566c677b16c9092f1a01538d07c608fb82b35897eda9ddb4R249-R252) by this PR
```

Reply:
Yes, this improvement will help a little.

Note that I was not clear enough in my comment. I understood that the naming had to evolve and "signal sighting" was not correct enough. I was just trying to highlight that maybe "signal critical position" felt like an unfinished sentence: I would expect something after like "signal critical position before the train has to stop" (which is obviously too long for a name): "critical position" seems to refer to something that is not described in the name.

I don't really have a good solution to propose, so I'd be fine with the small improvement you propose above (feel free to resolve). Yet, I'll propose some terms here (in brainstorming mode), just in case it triggers something in you: "signal distance to stop", "maximum distance before signal", "ultimate position before stop".

Reply:
Pushed (directly amended) the suggestion: https://github.com/OpenRailAssociation/osrd/compare/ecce330889fd3885491eeb20b66a2b154a747394..461433f4d9aa532fc76cb4eb04dffa6d1b7d0ba5

Not really triggered by the others terms, we may dig along the fact that non-free signal would be a (broken) constraint from that position for this train-schedule. But I think that the current name is OK as we had an understanding already (between signaling-system devs at least).

So I resolve, but I am OK if anybody unresolves to bring up something they really think is clearer.

Reply:
> "signal distance to stop", "maximum distance before signal", "ultimate position before stop"

I dislike all of those, sorry :/

It's really "spacetime point where the signal has to be green for no slowdown". It's not (only) about stops nor is it a distance, and the time component is more important. You can recover the spatial component from the time one if you have the enveloppe, you can't do it the other way - since it may be during a stop. (This is what I dislike most about "critical position", which come from a variable name in the routing ressource generation algorithm from before closed-signal stop support, and thus couldn't be a point during a stop)

Reply:
You have a point with the time vs. space, so `SignalCriticalTime` is better, then?
I'm OK to change this if you agree :)

Reply:
Oh, to be honest, I dislike all of them too. I was just throwing more new words to see if it might trigger new ideas from the experts 😅 It didn't, and that's fine, especially since

> But I think that the current name is OK as we had an understanding already (between signaling-system devs at least).

"critical position" seems to be a shared jargon term between expert in signaling-system, let's keep it like this.

Reply:
As discussed aside: not worth changing too much the scope of this PR, let's keep it to `critical_position`, even if `critical_time` is a valid idea.